### ISSUE TITLE
Layer properties are not accessible if its style is invalid

### ISSUE THREAD

**Issue by allyoucanmap** [geosolutions-it/MapStore2#3415]
### Description
In some cases styles are created and updated correctly but when apply to the layer return an exception.
If GetMap has exception TOCItemSettings is not accessible and the style parameters can't be changed.

* Following style cause the issue
```
@styleTitle 'Solid fill';
@styleAbstract '';

[this_is_not_a_param = 'a'] {
	fill: #aaaaaa;
}
```

### In case of Bug (otherwise remove this paragraph)

*Browser Affected*
(use this site: https://www.whatsmybrowser.org/ for non expert users)
- [ ] Internet Explorer
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

*Browser Version Affected*

- all

*Steps to reproduce*

- open a map with vector layer
- select the layer and open the settings of layer
- select style tab
- create a new style and edit it
- write the style* that cause the error

*Expected Result*

- There should be a way to access the parameters to update the layer parameters or at least reset the layer parameters to ensure that the error depends on wrong paramaters

*Current Result*

- Layer parameters can't be chaged and settings aren't accessible

**Comment by gioscarda** [comment]
I've tried it on geonode with the `states` layer.
I've created a new CSS style for this layer and then I edit it as suggested by @allyoucanmap.

The CSS style name is `My style`.

The style has been saved without errors, I returned back and saved the changes with the save button. After that the TOC appears like this:

![image.png](https://images.zenhubusercontent.com/5cda9c3818690954352b7b76/c10ae0b8-1cb8-4f02-9928-58207c72a95b)

So the layers settings are available but when I try to re-edit the style its name is changed in `Solid fill`:
![image.png](https://images.zenhubusercontent.com/5cda9c3818690954352b7b76/547143aa-daf9-43f3-9397-7982e4742b24)

Is this acceptable?
